code-challenges.test.js

```diff
@@ -1,67 +1,249 @@
+// 4.) append .sort() Object.value() to access the array and sort the value at index from least to greatest
+
+// */
+
+// const studyMinutesWeek1 = {
+//       sunday: 90,
+//       monday: 30,
+//       tuesday: 20,
+//       wednesday: 15,
+//       thursday: 30,
+//       friday: 15,
+//       saturday: 60
+//       }
+//       // Expected output: [15, 15, 20, 30, 30, 60, 90]
+//     const studyMinutesWeek2 = {
+//       sunday: 100,
+//       monday: 10,
+//       tuesday: 45,
+//       wednesday: 60,
+//       thursday: 20,
+//       friday: 15,
+//       saturday: 65
+//       }
+//       // Expected output: [10, 15, 20, 45, 60, 65, 100]
+
+ const smallBigArray = (study) => {
+  return Object.values(study).sort((a, b) => a - b)
+}
+
+/*
+
+PASS  ./code-challenges.test.js
+  smallBigArray
+    ✓ return an array of values sorted from least to greatest (1 ms)
+
+Test Suites: 1 passed, 1 total
+Tests:       1 passed, 1 total
+Snapshots:   0 total
+Time:        0.199 s
+
+    */
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const accountTransactions1 = [100, -17, -23, -9]
-// Expected output: [100, 83, 60, 51]
+describe("accumulatingSum", () => {
+  const accountTransactions1 = [100, -17, -23, -9]
+  // Expected output: [100, 83, 60, 51]
+
+  const accountTransactions2 = [250, -89, 100, -96]
+  // Expected output: [250, 161, 261, 165]
+
+  const accountTransactions3 = []
+  // Expected output: []
+    it("return an array of with the sum added from the pervious index", () => {
+    expect(accumulatingSum(accountTransactions1)).toEqual([100, 83, 60, 51])
+    expect(accumulatingSum(accountTransactions2)).toEqual([250, 161, 261, 165])
+    expect(accumulatingSum(accountTransactions3)).toEqual([])
+  })
+})
 
-const accountTransactions2 = [250, -89, 100, -96]
-// Expected output: [250, 161, 261, 165]
+/*
 
-const accountTransactions3 = []
-// Expected output: []
+FAIL  ./code-challenges.test.js
+accumulatingSum
+  ✕ return an array of with the sum added from the pervious index
+
+● accumulatingSum › return an array of with the sum added from the pervious index
+
+  ReferenceError: accumulatingSum is not defined
+
+*/
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 // b) Create the function that makes the test pass.
+
+/*
+
+Pseudocode: 1.) Create an function named accumulatingSum
+2.) accumulatingSum will take argument of an array data type and return the accumulating sum
+3.) Create a variable named accSum to store data from arithmetic
+4.) Use a reduce() to iterate over the indexes of the array
```

Comment:
OK!!! Love the thought here with reduce. Reduce can be a bit tricky but one of the things it is really useful for is an accumulating sum. I would challenge you to refactor this one now that you have it working and see if you can do it without a few things. 😉